diff --git a/bayesian-network.js b/bayesian-network.js
index 2532dc91a593bd414433c422e78613e2e56bfa5f..c9fb2fbe238193af85a804aa46d9c7afabca5183 100644
--- a/bayesian-network.js
+++ b/bayesian-network.js
@@ -2,6 +2,7 @@
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.BayesianNetwork = void 0;
 const AdmZip = require("adm-zip");
+const { readFileSync } = require("fs");
 const bayesian_node_1 = require("./bayesian-node");
 /**
  * BayesianNetwork is an implementation of a bayesian network capable of randomly sampling from the distribution
@@ -21,7 +22,7 @@ class BayesianNetwork {
             writable: true,
             value: {}
         });
-        const zip = new AdmZip(path);
+        const zip = new AdmZip(readFileSync(path));
         const zipEntries = zip.getEntries();
         const networkDefinition = JSON.parse(zipEntries[0].getData().toString('utf8'));
         this.nodesInSamplingOrder = networkDefinition.nodes.map((nodeDefinition) => new bayesian_node_1.BayesianNode(nodeDefinition));
